src/profile-logic/comparison.js

```diff
@@ -28,11 +28,11 @@ export function mergeCategories(
 |} {
   const newCategories = [];
   const translationMaps = [];
+  const insertedCategories: Map<string, IndexIntoCategoryList> = new Map();
 
   categoriesPerThread.forEach(categories => {
     const translationMap = new Map();
     translationMaps.push(translationMap);
-    const insertedCategories: Map<string, IndexIntoCategoryList> = new Map();
```

Comment:
Thanks to the change to use `serializeProfile` (which does `JSON.stringify` and and so we get a copy of the profile) this uncovered this important error. Previously I was mutating the profile object... that we were using also as "expected", so obviously the test was passing...